mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -712,6 +722,9 @@ class SparseVector @Since("1.0.0") (
     }
   }
 
+  @Since("1.6.0")
+  private[spark] def foreach(f: (Double) => Unit) = { indices.foreach { i: Int => f(values(i)) } }
```

Comment:
yah, I think this implementation was the result of a miscommunication, I believe the intended purpose is to provide something matching the normal foreach function (why the PR was changed back to a WIP)